### PR TITLE
ISO8583 Integration Deep Dive and Proposal

### PR DESCRIPTION
**Pull request by adit313**
# Milestone Delivery Checklist

- [x] The [milestone-delivery-template.md](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/deliveries/milestone-delivery-template.md) has been copied and updated.
- [x] The [invoice form :pencil:](https://forms.gle/LSRr7PCjBpEbKGh89) has been filled out for this milestone.
- [x] This pull request is being made by the same account as the accepted application.
- [x] In case of acceptance, the payment will be transferred to the BTC/ETH payment address provided in the application.
- [x] The delivery is according to the [Guidelines for Milestone Deliverables](https://github.com/w3f/Grants-Program/blob/master/docs/milestone-deliverables-guidelines.md).

Link to the application pull request: https://github.com/w3f/Grants-Program/pull/XXX < please fill this in with the PR number of your application.

### CODE REVIEW

**Comment by adit313** [comment]
Hi all! Just submitted the presentation on a deep dive into integrating ISO8583 as well as a deeper dive into the card payments industry it supports as I think a lot of that is very necessary for context. Sorry for the delay, there was quite a bit of material to research and cover for completeness. I tried to spare you the details as much as I could, though I've included all the links for anyone who is curious and wants more detail. I'm more than happy to append additional detail on any slide and build out anything of interest to the team.

Also, there is just a lot that comes with integrating ISO8583, so I'm also happy to jump on a call to debrief with the team and answer questions. I've tried to answer the highest level ones, but I'm sure you'll have a ton after reading this presentation. Also, let me know if I've used any terms you aren't familiar with, I've tried to be as explicit and limited with industry jargon as possible, but sometimes old habits die hard from my analyst days.

**Comment by takahser** [comment]
@adit313 thanks for the delivery and the additional information you provided in your [comment](https://github.com/w3f/Grant-Milestone-Delivery/pull/586#issuecomment-1264871949). I'm going to take a look at it and give you feedback soon.

**Review by takahser — CHANGES_REQUESTED** [review]
@adit313 could you add a LICENSE file to your [repository](https://github.com/adit313/ISO8583-Blockchain-Integration-Plan)? We [prefer](https://github.com/w3f/Grants-Program#guidelines) Apache 2.0, but GPLv3, MIT or Unlicense are also acceptable.

**Comment by adit313** [comment]
Of course! Just added an Apache 2.0 license to the repo.

Also, let me know if you need me to make any changes or explore any alternative paths. I made the assumption that your ultimate end goal for the ISO8583 project is to integrate with payment card networks and the proposed methodology is really designed to achieve that use case. I've never heard of another use for ISO8583 outside of supporting payment networks and can't really think of a reason you'd want to adopt the standard for general use, but let me know if there was another intended use case you had for ISO8583.

File: deliveries/ISO8583-milestone-1.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+- **Application Document:** [Application](https://github.com/w3f/Grants-Program/blob/master/applications/Integrating-ISO8583.md).
+- **Milestone Number:** 1
+
+**Context** (optional)
+
+This is the associated deliverable for milestone 1, a well-detailed analysis and implementation plan for incorporating ISO8583 with Substrate based blockchains. ISO8583 is the worldwide payment card standard, however due to the fact that it encodes sensative financial information such as personal information and card numbers in plain-text, it is a non-trivial matter to support it on any public blockchain. This presentation highlights the major challenges and reccomends an oracle based implementation to comply with payment card network's terms of service as well as payment card regulations. Note: All sources are cited in the notes section of the PowerPoint under each slide, our exported PDF contains only the final slide data.
```
Comment (takahser) [inline deliveries/ISO8583-milestone-1.md:10]:
```suggestion
This is the associated deliverable for milestone 1, a well-detailed analysis and implementation plan for incorporating ISO8583 with Substrate based blockchains. ISO8583 is the worldwide payment card standard, however, due to the fact that it encodes sensitive financial information such as personal information and card numbers in plain text, it is a non-trivial matter to support it on any public blockchain. This presentation highlights the major challenges and recommends an oracle based implementation to comply with payment card network's terms of service as well as payment card regulations. Note: All sources are cited in the notes section of the PowerPoint under each slide, our exported PDF contains only the final slide data.
```

**Review by takahser — CHANGES_REQUESTED** [review]

**Review by takahser — APPROVED** [review]
@adit313 thanks for your patience. It took me a while to read through your research but I learned a lot and I'm happy with its coverage. It seems to me, like your suggested architecture would be feasible both from a technical as also a legal perspective. Is this something you'd be interested to pursue further?
Also, I approved your delivery, you can find my evaluation [here](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/evaluations/iso8583-integration_1_takahser.md).

**Comment by adit313** [comment]
Definitely interested in pursuing it further. Using any crypto day-to-day is a big pain point for me. I normally have to rely on exchanges and go through all of the logistics of converting to fiat, transferring to a bank, etc. It would also have MASSIVE monetary advantages if something like the [Virtual Currency Tax Fairness Act of 2022](https://www.congress.gov/bill/117th-congress/house-bill/6582/text) were to pass and exempt any crypto transaction under $50 from taxes. I kind of selfishly want it so it's easier for me to use it through just a payment card.

In terms of roadmap, I designed it in such a way that it is relatively modular and can be prototyped and built in pieces. I'm thinking it makes sense to create the ISO8583 accepting backend and oracle first, followed by building a practical implementation of that ERC20R smart contract proposed by Stanford on Substrate. That way we can build in the oracle based authorizations and reversals, then test them with the active Oracle. This would then be followed by setting up a real connection to the payment networks (Visa/MC) through one of those banking-as-a-service BIN sponsors and getting a few physical test cards issued so we can test the functionality. It's more involved than that obviously, but I've already scoped out the technologies and it all looks feasible. 

I know several members of the team are also super interested in this, and I was thinking it might make sense at this stage to also apply for the Substrate Builders Program. That way we can have a much tighter feedback loop and regular check-ins. Curious to get your thoughts on if it's a good fit for that program at this time.

**Comment by Noc2** [comment]
@adit313 Great to hear! Just to mention it: Besides the SBP, we are also interested in supporting this via our grants program, as long as we don’t double fund the same work. Feel free to apply for another grant.

**Comment by adit313** [comment]
Thanks David! I'll be sure to apply for another grant in parallel. I actually assumed the SBP didn't come with any funding, I thought it was mostly for higher level strategic and technical partnership and support. I'll make sure to tell them that I am applying for a grant here.

**Comment by adit313** [comment]
Hi all, it's been a while so I wanted to give an update as to why progress is so slow and taking so long. I've also received a Phase I SBIR award grant from the U.S.'s National Science Foundation in Distributed Ledger Technology so I've been swamped with other research work. I just haven't really had time to sit down and focus on this. I just wanted to apologize and to keep you guys in the loop.

**Pull request merged**